### PR TITLE
Polish dashboard and insights hierarchy

### LINKED ISSUE #59: Polish dashboard hierarchy, chart readability, and category-progress presentation across shipped screens
## Why it matters  
The app already has attractive screens, so targeted polish here has outsized payoff. This is where the project starts to look truly intentional rather than merely functional.
## Scope  
Run a focused polish pass on the highest-value UI areas: dashboard card hierarchy and spacing, chart labels and markers, chart contrast and readability, budget-state visibility, more glanceable budget danger cues, category-progress presentation, and mobile spacing and interaction quality across Dashboard, Insights, and Transactions.
## Acceptance Criteria  
- Core charts and budget modules are easy to read at a glance.
- Budget and category status are visually clearer than they are now.
- Dashboard hierarchy makes key status more obvious than secondary details.
- Dangerous budget states are more visually immediate and less text-heavy than they are now.
- Mobile layout for dashboard, insights, and transactions feels intentional.
- No high-visibility chart or card looks confusing or unfinished in the final presentation.
## Story Points  
3

### PR DESCRIPTION
**Pull request by adillpickles**
## Description
Polishes the Dashboard, Insights, and Transactions presentation for Issue #59.
This pass improves visual hierarchy, chart readability, and category-progress presentation by flattening the Dashboard hero, standardizing shared UI primitives, improving pacing/cash-flow/category visuals, and adding focused drill-down/detail interactions. It also adds and updates frontend coverage for the new UI components and budget-health states, including a new “watch” threshold.

## Related User Story / Issue

Fixes #59

## Type of Change

- [ ] New feature
- [ ] Bug fix
- [ ] Backend / API change
- [x] UI change
- [ ] Database change
- [ ] Documentation
- [x] Refactor

## How Has This Been Tested?

- [ ] GitHub Actions / CI tests passed
- [x] Manual testing performed
- [ ] Not tested locally

### Test Notes

Local verification:
- `npx jest --no-coverage` — 28 suites / 307 tests passed
- `npm run build` — production build succeeded
- Manual browser review of Dashboard, Insights, Planner, and Transactions

## UI Changes (if applicable)

- [ ] No UI changes
- [x] UI updated (add screenshots if needed)

## Screenshots (if UI changed)
image
image
image
image
image
image

## Notes for Reviewers

- Large UI/CSS polish pass scoped to Issue #59.
- Adds shared UI primitives for allocation bars, category progress rows, pacing charts, financial health, cash-flow snapshots, transaction detail display, and category drill-downs.
- Most new styling is scoped to the Issue #59 section/classes in `nextjs/src/app/globals.css`.
- Adds a new `watch` budget health threshold at `0.6` to support softer warning states before near-limit/over-budget states.
- No package, CI, auth, Planner, or release-workflow changes are intended in this PR.

Follow-up polish that may be worth a separate issue/PR:
- The Transactions floating `+` button can overlap the bottom nav at some viewport sizes and should be repositioned or made nav-aware. Did not fix that in this PR. 
- Planner could eventually reuse the Insights-style month modal / month switcher interaction.
- The month-view modal could support previous/next month navigation.
- Some chart/status color tuning may still be refined later, but this PR establishes the shared visual primitives and consistency layer.

## Checklist

- [x] Linked to a user story or issue
- [x] Code builds / tests pass locally
- [x] Ready for review

### CODE REVIEW

File: nextjs/e2e/auth.spec.js
Comment (zakariac22) [inline nextjs/e2e/auth.spec.js:1]: Since the other E2E test file transaction.spec.js enters the tester credentials as variables, you can do the same in this file for consistency.

**Comment by zakariac22** [comment]
Screenshot 2026-04-25 at 11 31 43 AM
For all the new added components, let's have a target coverage of 90% or greater, as this will be the standard for release.

File: nextjs/__tests__/frontend/transactions.test.js
Comment (zakariac22) [inline nextjs/__tests__/frontend/transactions.test.js:1]: You should make a test case here for backdrop close (clicking the background area). The overlay has a separate backdrop button with onClick={onClose} but it only tests the labeled "Close" button.

File: nextjs/__tests__/frontend/transactions.test.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    render(React.createElement(TransactionsView))
+    act(() => {
+      fireEvent.click(screen.getByText('Mockville Payroll'))
+    })
+    const subtitle = document.querySelector('.detail-sheet__subtitle')
+    expect(subtitle).toBeTruthy()
+    expect(subtitle.textContent).not.toBe('Income')
+  })
+
+  it('shows a kind-specific empty message when filters produce no results', () => {
```
Comment (zakariac22) [inline nextjs/__tests__/frontend/transactions.test.js:180]: The type of label for this test can be more narrowed, as only one search query is asserted as an expected outcome, and no test case for other search queries. You can either narrow the name of the test or expand the query cases.

File: nextjs/__tests__/frontend/financeUtils.unit.test.js
Comment (zakariac22) [inline nextjs/__tests__/frontend/financeUtils.unit.test.js:1]: I'd recommend adding another edge case such as a test that verifies the fallback IDs are unique when there are a few rows from the same day.

File: nextjs/e2e/auth.spec.js
Comment (adillpickles) [reply nextjs/e2e/auth.spec.js:1]: updated `auth.spec.js` to use `TEST_EMAIL` / `TEST_PASSWORD` constants, matching the style used in the transactions E2E spec.

File: nextjs/__tests__/frontend/transactions.test.js
Comment (adillpickles) [reply nextjs/__tests__/frontend/transactions.test.js:1]: for coverage comment above targeted coverage for the new `src/components/ui` files is now around/above 90% statements
and for backdrop close added a transaction detail sheet test that opens the sheet and closes it through the backdrop’s accessible close control, not only the labeled close button

File: nextjs/__tests__/frontend/transactions.test.js (hunk 1 above)
Comment (adillpickles) [reply nextjs/__tests__/frontend/transactions.test.js:180]: narrowed the test name so it accurately describes the single search query being asserted instead of implying broader search coverage

File: nextjs/__tests__/frontend/financeUtils.unit.test.js
Comment (adillpickles) [reply nextjs/__tests__/frontend/financeUtils.unit.test.js:1]: added a focused `financeUtils` unit test covering multiple same day rows without stable IDs and verifying the fallback IDs remain unique

File: nextjs/src/components/dashboard-view.js [hunk 2]
```diff
@@ -334,7 +263,11 @@ export function getBudgetPressureHighlight(summary, expenses = [], derivedCatego
...
   return buildSharedBudgetPressureHighlight({
     categoryStatuses: summary?.category_statuses,
-    fallbackSpendCards: spendShareCards,
+    fallbackSpendCards: spendShareCards.map((card) => ({
+      name: card.name,
+      note: card.note,
+      progress: card.progress,
+    })),
   })
 }
```
[the remark this answers is not in the document]
Comment (adillpickles) [reply nextjs/src/components/dashboard-view.js:273]: good catch, glad the bot is working. i preserved amount in the fallback spend card mapping so the pressure highlight can keep the spent value

**Review by zakariac22 — APPROVED** [review]
